iroha_p2p/src/peer.rs

```diff
@@ -403,13 +476,122 @@ impl Garbage {
 
     pub async fn read(stream: &mut OwnedReadHalf) -> Result<(), Error> {
         let size = stream.read_u8().await? as usize;
-        if size < MAX_HANDSHAKE_LENGTH {
-            // Reading garbage
-            let mut buf = vec![0_u8; size];
-            let _ = stream.read_exact(&mut buf).await?;
-        } else {
+        if size >= MAX_HANDSHAKE_LENGTH {
             return Err(Error::Handshake);
         }
+        // Reading garbage
+        let mut buf = vec![0_u8; size];
```

Comment:
Btw, may be we can preallocate buffer of maximum size by default for each pear?

Reply:
Preallocate 256 bytes and never free? I don't like this idea :(

Reply:
I guess it can be done in other pr with typestates. It memory reclaimation will be clearer there.

Reply:
Can i take it though? I feel like that would be way easier and faster to do me as i have already some clear image of it.

Reply:
Yep, you can do it :)